golden/golden.go

```diff
@@ -42,7 +43,8 @@ func Path(filename string) string {
 
 func update(filename string, actual []byte) error {
 	if *flagUpdate {
-		return ioutil.WriteFile(Path(filename), actual, 0644)
+		s := strings.Replace(string(actual), "\r", "", -1)
```

Comment:
This can be done directly with bytes instead of converting to a string and back to bytes: https://golang.org/pkg/bytes/#Replace

I think we should make 2 small changes here:
1. It could use `bytes.Replace(actual, []byte("\r\n"), []byte("\n"))` so that it only replaces `\r` when it is followed by `\n`
2. it should probably only replace when called from `String()`. If someone is using `Bytes` or `AssertBytes` they might be dealing with binary data, so we shouldn't replace the `\r`.